hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/evm/account/MirrorEvmContractAliases.java

```diff
@@ -78,6 +78,11 @@ public Address resolveForEvm(Address addressOrAlias) {
         }
     }
 
+    public boolean isNativePrecompileAddress(Address addressOrAlias) {
+        final var addressDecimals = Long.parseLong(addressOrAlias.toHexString().substring(2), 16);
+        return addressDecimals > 0 && addressDecimals < 10;
+    }
```

Comment:
This narrowing conversion to Long could result in an exception for high numbered addresses. Safer to do comparison using address itself.

```suggestion
    public boolean isNativePrecompileAddress(Address address) {
        return address != null && address.compareTo(Address.ZERO) > 0 && address.compareTo(TEN) < 0;
    }
```

Reply:
You're right, yes. Fixed.